From f1fe9e4487377ccd481226d7074e88e794ceadd2 Mon Sep 17 00:00:00 2001
From: Bilal815 <Bilalahmed815@gmail.com>
Date: Sat, 29 Aug 2020 16:49:31 +0500
Subject: [PATCH] Create Patch File

---
 client3.py     |  8 +++-----
 client_test.py | 26 ++++++++++++++++++++++++--
 2 files changed, 27 insertions(+), 7 deletions(-)

diff --git a/client3.py b/client3.py
index 9bc0c7a..d7ae34c 100644
--- a/client3.py
+++ b/client3.py
@@ -42,7 +42,6 @@ def getRatio(price_a, price_b):
 	""" Get ratio of price_a and price_b """
 	""" ------------- Update this function ------------- """
 	""" Also create some unit tests for this function in client_test.py """
-	return 1
 	if (price_b == 0):
     		return
 	return price_a/price_b
@@ -58,8 +57,7 @@ if __name__ == "__main__":
 		price = {}
 		for quote in quotes:
 			stock, bid_price, ask_price, price = getDataPoint(quote)
-      		print ("Quoted %s at (bid:%s, ask:%s, price:%s)" % (stock, bid_price, ask_price, price))
 			prices[stock] = price
-   				print ("Quoted %s at (bid:%s, ask:%s, price:%s)" % (stock, bid_price, ask_price, price))
-		
-  		print ("Ratio %s" % getRatio(price, price))
+   			print ("Quoted %s at (bid:%s, ask:%s, price:%s)" % (stock, bid_price, ask_price, price))
+       
+  		print ("Ratio %s" % getRatio(prices['ABC'], prices['DEF']))
diff --git a/client_test.py b/client_test.py
index 6791ea5..9a283b6 100644
--- a/client_test.py
+++ b/client_test.py
@@ -21,8 +21,30 @@ class ClientTest(unittest.TestCase):
           self.assertEqual(getDataPoint(quote), (quote['stock'], quote['top_bid']['price'], quote['top_ask']['price'], (quote['top_bid']['price'] + quote['top_ask']['price'])/2))
 
   """ ------------ Add more unit tests ------------ """
-    for quote in quotes:
-          self.assertEqual(getRatios(quote), ( quote['stock'], quote['top_bid']['price'], quote['top_ask']['price'], (quote['top_bid']['price']/quote['top_ask']['price'])))
+  def test_getRatio_priceBZero(self):
+    price_a = 119.2
+    price_b = 0
+    self.assertIsNone(getRatio(price_a, price_b))
+
+  def test_getRatio_priceAZero(self):
+    price_a = 0
+    price_b = 121.68
+    self.assertEqual(getRatio(price_a, price_b), 0)
+ 
+  def test_getRatio_greaterThan1(self):
+    price_a = 346.48
+    price_b = 166.39
+    self.assertGreater(getRatio(price_a, price_b), 1)
+
+  def test_getRatio_LessThan1(self):
+    price_a = 166.39
+    price_b = 356.48
+    self.assertLess(getRatio(price_a, price_b), 1)
+
+  def test_getRatio_exactlyOne(self):
+    price_a = 356.48
+    price_b = 356.48
+    self.assertEqual(getRatio(price_a, price_b), 1)
 
 
 if __name__ == '__main__':
-- 
2.27.0.windows.1

